alf/metrics/metric.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2020 Horizon Robotics. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""A few metrics.
+Code adapted from https://github.com/tensorflow/agents/blob/master/tf_agents/metrics/tf_metric.py
+"""
+
+import alf
+import os
+
+from torch import nn
+
+
+class StepMetric(nn.Module):
+    """Defines the interface for metrics."""
+
+    def __init__(self, name, prefix='Metrics'):
+        super().__init__()
+        self.name = name
+        self._prefix = prefix
+
+    def __call__(self, *args, **kwargs):
+        return self.call(*args, **kwargs)
+
+    def call(self, *args, **kwargs):
+        """Accumulates statistics for the metric.
+    Args:
+      *args:
+      **kwargs: A mini-batch of inputs to the Metric.
+    """
```

Comment:
Docstring format needs to be fixed. It seems that locally you didn't run pre-commit hooks before pushing.

Reply:
Fixed.  I upgraded tf but forgot to install hooks.  Done it now.